doc/rst/search_index_entries.py

```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*-
+"""
+    search_index_entries.py
+    ~~~~~~~~~~~~~~~~~~~~~~~
+
+    This file generates search-index-entries.js which can be used from
+    the javascript side in order to more prominently feature search
+    results that match index items.
+
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import json
+import os.path
+from docutils import nodes
+from sphinx.addnodes import index
+from sphinx.util import logging
+
+if TYPE_CHECKING:
+    from sphinx.application import Sphinx
+
+logger = logging.getLogger(__name__)
+JSON = 'searchindex.js'
+
+# For Python 3.9 and later, just use removeprefix / removesuffix
+def remove_prefix(text, prefix):
+    if text.startswith(prefix):
+        return text[len(prefix):]
+    return text
+
+def remove_suffix(text, suffix):
+    if text.endswith(suffix):
+        return text[:-len(suffix)]
+    return text
+
+def process_index_nodes(app: Sphinx, doctree: nodes.document, docname: str):
+    if app.builder.format != 'html':
+        return
+
+    terms = { }
+
+    for node in doctree.traverse(index):
+        for i, entry in enumerate(node['entries']):
+            (entrytype, entryname, target, _, _) = node['entries'][i]
```

Comment:
There isn't a need to use the `enumerate` function here.

```suggestion
for entry in node['entries']:
    (_, entryname, target, _, _) = entry
```
Alsom `entrytype` is unused in the loop body and can be replaced with `_`